### PR TITLE
Update Go to 1.18

### PR DESCRIPTION
**Pull request by tklauser**
Also switch CI and docs from `go get -u ...` to `go install` as the former is deprecated as of Go 1.17 and no longer supported on Go 1.18 (see https://go.dev/doc/go-get-install-deprecation)

Also change how `golangci-lint` is integrated in the Makefiles and GitHub actions.

See commits for details.

### CODE REVIEW

**Review by ldelossa — APPROVED** [review]
approved contingent on tests ✅

**Review by rolinh — APPROVED** [review]

**Review by rolinh — CHANGES_REQUESTED** [review]
Running `go fix ./...` allows to get rid of the now useless `// +build` directives in all the files and only keep the `//go:build` ones.

**Review by rolinh — APPROVED** [review]

**Review by christarazi — APPROVED** [review]
:tada:

**Review by sayboras — APPROVED** [review]
Nice 🚀

File: Makefile
Comment (qmonnet) [inline Makefile]: Can we keep some documentation for the `lint` target? I think the comment after the `##` is parsed and displayed for `make help`.

File: .github/workflows/tests-smoke.yaml
Comment (qmonnet) [inline .github/workflows/tests-smoke.yaml]: Should there be a comment that points to the Prometheus issue, so that we can restore simpler steps once it's addressed?

**Review by nebril — APPROVED** [review]

**Review by gandro — APPROVED** [review]

File: .github/workflows/tests-smoke.yaml
Comment (aanm) [inline .github/workflows/tests-smoke.yaml]: Why `2.31.1`? Perhaps we could update to `v2.34.0` and use the pinned commit `881111fec4332c33094a6fb2680c71fffc427275`?

File: .github/workflows/lint-build-commits.yaml
Comment (aanm) [inline .github/workflows/lint-build-commits.yaml]: Let's pin the commit sha while we are at it `cc0216944b25a88d3259699a029d4e601fb8a222`

File: .travis/prepare.sh
Comment (aanm) [inline .travis/prepare.sh]: Accordingly the docs, this tools is now part of the std library, should we remove it?

File: .travis/prepare.sh
Comment (aanm) [inline .travis/prepare.sh]: Let's pin it to a commit `a36c7ef8f23b2952fa6e39663f52107dfc8ad69d` [v0.0.11](https://github.com/mattn/goveralls/releases/tag/v0.0.11)

File: test/consul/gen-cert.sh
Comment (aanm) [inline test/consul/gen-cert.sh:21]: Let's pin to `29ae05fe80e1a9c704ddad7002d90ade7a38cb29` which is `1.6.1`

File: test/consul/gen-cert.sh
Comment (aanm) [inline test/consul/gen-cert.sh:28]: Let's pin to `29ae05fe80e1a9c704ddad7002d90ade7a38cb29` which is `1.6.1`

File: test/packet/scripts/install.sh
Comment (aanm) [inline test/packet/scripts/install.sh]: `db6b6e9ba07629c90654a7720c452fb44a864dfb` which is `v0.3.22`

File: test/packet/scripts/install.sh
Comment (aanm) [inline test/packet/scripts/install.sh]: `1ed6486197ba2a007858c59bdc4d1522631025d5` which is `v2.1.3`

File: tools/dev-doctor/rootcmd.go
Comment (aanm) [inline tools/dev-doctor/rootcmd.go:93]: `1ed6486197ba2a007858c59bdc4d1522631025d5`

File: Makefile
Comment (aanm) [inline Makefile]: `docker.io/golangci/golangci-lint:v$(GOLANGCILINT_WANT_VERSION)@$(GOLANGCILINT_IMAGE_SHA)`

File: Makefile.defs
Comment (aanm) [inline Makefile.defs]:
```
GOLANGCILINT_IMAGE_SHA = sha256:c61216e0aeb8b38dcdcf790e2b7eee6e2f8c5564bd5f938ce09f7a027a409d56
```

(this SHA is for the v1.45.0 so it can be added in the next commit)

File: .github/workflows/tests-smoke.yaml
Comment (aanm) [inline .github/workflows/tests-smoke.yaml:191]: Small nit, there's a `https://github.com/prometheus/prometheus/releases/download/v2.34.0/sha256sums.txt` that we can use to verify prometheus binaries.

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by qmonnet — APPROVED** [review]

File: Makefile.defs
Comment (rolinh) [reply Makefile.defs]: v1.45.2 is already out.

File: .github/workflows/tests-smoke.yaml
Comment (tklauser) [reply .github/workflows/tests-smoke.yaml]: This is updated to v2.34.0 as part of the switch to the binary release in the second to last commit. I'll keep the existing version in this commit.

File: Makefile.defs
Comment (tklauser) [reply Makefile.defs]: Updated to v1.45.2 (SHA `e84b639c061c8888be91939c78dae9b1525359954e405ab0d9868a46861bd21b`)

**Review by qmonnet — APPROVED** [review]
Approving again, because I like it. :tophat:

**Comment by tklauser** [comment]
Rebased again to pick up https://github.com/cilium/cilium/pull/19230

**Comment by tklauser** [comment]
/test-1.22-4.19

**Review by ti-mo — APPROVED** [review]

**Comment by tklauser** [comment]
Tests passed, review comments are addressed and several reviewers approved. Most of the review requests got added for simple `// +build`/`//go:build` changes in their respective packages, so there seems to be no need to wait for approval for each of them. Marking as ready to merge.

**Comment by qmonnet** [comment]
@tklauser should [`PRIV_TEST_PKGS_EVAL` in the main Makefile](https://github.com/cilium/cilium/blob/9a7bd38c82c94a31b71fff7e0c1938634af8c3d4/Makefile#L133) have been updated after removing the `+build` tags? `make` complains on my setup, I believe because of that change:

```
$  make help > /dev/null
dirname: missing operand
Try 'dirname --help' for more information.
```

**Comment by tklauser** [comment]
> @tklauser should [`PRIV_TEST_PKGS_EVAL` in the main Makefile](https://github.com/cilium/cilium/blob/9a7bd38c82c94a31b71fff7e0c1938634af8c3d4/Makefile#L133) have been updated after removing the `+build` tags? `make` complains on my setup, I believe because of that change:
> 
> ```
> $  make help > /dev/null
> dirname: missing operand
> Try 'dirname --help' for more information.
> ```

Good catch, I forgot to update this as part of this PR. Done now in #19415. Thanks.

**Pull request merged**
